packages/quantic/force-app/main/default/lwc/quanticModal/__tests__/quanticModal.test.js

```diff
@@ -0,0 +1,193 @@
+        () => mockFirstFocusableElement
+      );
+
+      const element = createTestComponent();
+      await flushPromises();
+
+      const headerSlot = element.shadowRoot.querySelector('slot[name=header]');
+      const contentSlot =
+        element.shadowRoot.querySelector('slot[name=content]');
+      const footerSlot = element.shadowRoot.querySelector('slot[name=footer]');
+
+      const headerSlotSpy = jest
+        .spyOn(headerSlot, 'assignedElements')
+        .mockImplementation(() => [document.createElement('div')]);
+      const contentSlotSpy = jest
+        .spyOn(contentSlot, 'assignedElements')
+        .mockImplementation(() => [document.createElement('div')]);
+      const footerSlotSpy = jest
+        .spyOn(footerSlot, 'assignedElements')
+        .mockImplementation(() => [document.createElement('div')]);
+
+      element.openModal();
+      await new Promise((resolve) => setTimeout(resolve, 600)); // jest.runAllTimers does not work
+
+      expect(getFirstFocusableElement).toHaveBeenCalled();
+      expect(mockFirstFocusableElement.focus).toHaveBeenCalled();
+
+      headerSlotSpy.mockRestore();
+      contentSlotSpy.mockRestore();
+      footerSlotSpy.mockRestore();
+    });
+  });
+
+  describe('when using slide to left animation', () => {
+    it('should have slide-to-left animation class', async () => {
+      const element = createTestComponent({animation: 'slideToLeft'});
+      await flushPromises();
+
+      const modal = element.shadowRoot.querySelector(selectors.modal);
+      expect(modal.classList.contains('hidden-modal_slide-to-left')).toBe(true);
+    });
+
+    it('should display the modal with slide-to-left animation when opened', async () => {
+      const element = createTestComponent({animation: 'slideToLeft'});
+      await flushPromises();
+      element.openModal();
+      await flushPromises();
+
+      const modal = element.shadowRoot.querySelector(selectors.modal);
+      expect(modal.classList.contains('modal_hidden')).toBe(false);
+      expect(modal.classList.contains('part-screen')).toBe(true);
+    });
+  });
+
+  describe('when fullScreen is set to true', () => {
+    it('should apply full-screen class to the modal', async () => {
+      const element = createTestComponent({fullScreen: true});
+      await flushPromises();
+      const modal = element.shadowRoot.querySelector(selectors.modal);
+      expect(modal.classList.contains('full-screen')).toBe(true);
+    });
+
+    it('should display the modal in full screen when opened', async () => {
+      const element = createTestComponent({fullScreen: true});
+      await flushPromises();
+      element.openModal();
+      await flushPromises();
+      const modal = element.shadowRoot.querySelector(selectors.modal);
+      expect(modal.classList.contains('modal_hidden')).toBe(false);
+      expect(modal.classList.contains('full-screen')).toBe(true);
+    });
+  });
+
+  describe('when setting an invalid animation', () => {
```

Comment:
We also have a logic that checks if the a valid value for the property `full-screen` has been passed, we could also test that.